Library/Homebrew/download_strategy.rb

```diff
@@ -555,24 +557,93 @@ def curl(*args, **options)
   end
 end
 
-# Strategy for downloading a file from an GitHub Packages URL.
+# Strategy for downloading a file from a Docker registry.
 #
 # @api public
-class CurlGitHubPackagesDownloadStrategy < CurlDownloadStrategy
```

Comment:
Would like to keep around a `CurlGitHubPackagesDownloadStrategy` or `GitHubPackagesDownloadStrategy ` subclass that avoids all need for authentication

Reply:
Can do that, but the current implementation is not spec-compliant and could break if Github makes their registry expect clients to be compliant. This change does still work fine with Github's registry.

Reply:
That change will not be made to GitHub's registry (I work there). I'd rather avoid the extra requests for the (vastly) more common case, thanks.